reference/sql_workarounds.rst

```diff
@@ -0,0 +1,141 @@
+    from dimensions
+   where fact_id in (select id from facts where ...conditions...)
+     and ...conditions...;
+
+Citus does not allow subqueries in the WHERE clause so we must choose a workaround.
+
+Workaround 1. Generate explicit WHERE-IN expression
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Pros
+    * Able to distribute the query across shards
+* Cons
+    * Constructed query not suitable when facts subquery returns large number of results
+    * Somewhat hard to read, requires building custom SQL
+
+In this technique we use PL/pgSQL to construct and execute one statement based on the results of another.
+
+.. code-block:: sql
+
+  -- create temporary table with results
+  do language plpgsql $$
+    declare facts_array integer[];
+  begin 
+    execute 'select id from facts where ...conditions...'
+       into facts_array;
+    execute format(
+      'create temp table results_temp as '
+      'select dim_a, dim_b, dim_c'
+      '  from dimension'
+      ' where fact_id = any(array[%s])'
+      '   and ...conditions...',
+      array_to_string(facts_array, ','));
+  end;
+  $$;
+
+  -- read results, remove temp table
+  select * from results_temp;
+  drop table results_temp;
+
+
+Workaround 2. Join on Master
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Pros
+    * Works for subqueries returning any number of results
+* Cons
+    * Must transmit full rows from the queries back to the master
+    * Requires extra storage space in the master database
+
+In this workaround the client runs the outer- and sub-query independently, saves their results, and joins them.
+
+.. code-block:: sql
+
+  -- Capture the dimension query results
+  create temp table dim_temp as
+  select fact_id, dim_a, dim_b, dim_c
+    from dimensions
+   where ...conditions...;
+  
+  -- Capture the subquery results
+  create temp table fact_temp as
+  select id
+    from facts
+   where ...conditions...;
+  
+  -- Join them, get filtered dimension results
+  select dim_a, dim_b, dim_c
+    from dim_temp, fact_temp
+   where dim_temp.fact_id = fact_temp.id;
+
+  -- Remove temp tables
+  drop table dim_temp;
+  drop table fact_temp;
+
+SELECT ... INTO
+---------------
+
+Citus does not support inserting the results of a query into a distributed table with the standard SELECT INTO command. One workaround is to use two database connections to stream the query results to master and then distribute them to the shards.
+
+.. code-block:: bash
+
+  psql -c "COPY (query) TO STDOUT" | psql -c "COPY table FROM STDIN"
+
+This does incur network cost. If this workaround is too slow please contact Citus Data support. We can assist you in parallelizing the table insertion across all workers using a more complicated technique.
+
+SELECT DISTINCT (non-distribution column)
+-----------------------------------------
+
+Citus supports SELECT DISTINCT col but only when col is the shard distribution column. If it is not, use GROUP BY for a simple workaround:
```

Comment:
SELECT DISTINCT col is not supported even when col is the distribution column. Could you correct this?